go/ql/src/experimental/CWE-770/DenialOfService.ql

```diff
@@ -0,0 +1,64 @@
+/**
+ * @name Denial Of Service
+ * @description slices created with the built-in make function from user-controlled sources using a
+ * maliciously large value possibly leading to a denial of service.
+ * @kind path-problem
+ * @problem.severity high
+ * @security-severity 9
+ * @id go/denial-of-service
+ * @tags security
+ *       experimental
+ *       external/cwe/cwe-770
+ */
+
+import go
+
+class BuiltInMake extends DataFlow::Node {
+  BuiltInMake() { this = Builtin::make().getACall().getArgument(0) }
+}
+
+/**
+ * Holds if `g` is a barrier-guard which checks `e` is nonzero on `branch`.
+ */
+predicate denialOfServiceSanitizerGuard(DataFlow::Node g, Expr e, boolean branch) {
```

Comment:
For every relational comparison node like `sink < 0` there is an instance of this class with `e = sink` and an instance with `e = 0`. I suggest that you add something like `not e.isConst()` (as there is no point sanitizing a constant expression). This will produce much fewer instances of the class and have the same results.

Reply:
I'll add `not e.isConst()`  condition to e